stripe/src/main/java/com/stripe/android/EphemeralKey.kt

```diff
@@ -0,0 +1,161 @@
+import android.os.Parcelable
+import com.stripe.android.model.StripeModel
+import java.util.Objects
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Represents an Ephemeral Key that can be used temporarily for API operations that typically
+ * require a secret key.
+ *
+ * See [
+ * Using Android Standard UI Components - Prepare your API](https://stripe.com/docs/mobile/android/standard#prepare-your-api) for more details on ephemeral keys.
+ */
+internal abstract class EphemeralKey : StripeModel, Parcelable {
+    val objectId: String
+    val created: Long
+    val expires: Long
+    val id: String
+    val isLiveMode: Boolean
+    val objectType: String
+    val secret: String
+    val type: String
+
+    /**
+     * Parcel constructor for this [Parcelable] class.
+     * Note that if you change the order of these read values,
+     * you must change the order in which they are written in
+     * [.writeToParcel] below.
+     *
+     * @param `in` the [Parcel] in which this Ephemeral Key has been stored.
+     */
+    constructor(parcel: Parcel) {
+        created = parcel.readLong()
+        objectId = Objects.requireNonNull<String>(parcel.readString())
+        expires = parcel.readLong()
+        id = Objects.requireNonNull<String>(parcel.readString())
+        isLiveMode = parcel.readInt() == 1
+        objectType = Objects.requireNonNull<String>(parcel.readString())
+        secret = Objects.requireNonNull<String>(parcel.readString())
+        type = Objects.requireNonNull<String>(parcel.readString())
+    }
+
+    constructor(
+        created: Long,
+        objectId: String,
+        expires: Long,
+        id: String,
+        liveMode: Boolean,
+        objectType: String,
+        secret: String,
+        type: String
+    ) {
+        this.created = created
+        this.objectId = objectId
+        this.expires = expires
+        this.id = id
+        isLiveMode = liveMode
+        this.objectType = objectType
+        this.secret = secret
+        this.type = type
+    }
+
+    override fun describeContents(): Int {
+        return 0
+    }
+
+    /**
+     * Write the object into a [Parcel]. Note that if the order of these write operations is
+     * changed, an identical change must be made to [EphemeralKey] constructor above.
+     *
+     * @param out a [Parcel] into which to write this object
+     * @param flags any flags (unused) for writing this object
+     */
+    override fun writeToParcel(out: Parcel, flags: Int) {
+        out.writeLong(created)
+        out.writeString(objectId)
+        out.writeLong(expires)
+        out.writeString(id)
+        // There is no writeBoolean
+        out.writeInt(if (isLiveMode) 1 else 0)
+        out.writeString(objectType)
+        out.writeString(secret)
+        out.writeString(type)
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) {
+            return true
+        }
+
+        return if (other is EphemeralKey) {
```

Comment:
method can be one line -
`return this === other || (other is EphemeralKey && typeEquals(other))`

Reply:
I don’t think the is check does smart casting with &&, only with if